OpenSim/Common/Component.h

```diff
@@ -2487,11 +2501,18 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     // subsystem.
     SimTK::ResetOnCopy<SimTK::MeasureIndex> _simTKcomponentIndex;
 
+    // list of subcomponents that are contained in this Component's properties
+    SimTK::Array_<SimTK::ReferencePtr<Component> >  _propertySubcomponents;
     // Keep fixed list of data member Components upon construction
     SimTK::Array_<SimTK::ClonePtr<Component> > _memberSubcomponents;
     // Hold onto adopted components
     SimTK::Array_<SimTK::ClonePtr<Component> > _adoptedSubcomponents;
 
+    // A flat list of all the subcomponents that comprise this Component.
+    // The list must be complete and in the correct System order by 
+    // addToSystem().
+    mutable std::vector<SimTK::ReferencePtr<const Component> > _orderedSubcomponents;
```

Comment:
`_orderedSubcomponents` is just a way for a Component to dictate the order in which subcomponents get added to the System. If the Component writer has no requirement on order then the list is auto generated in ownership traversal order. It includes all subcomponents regardless of internal designation (member, property, adopted)